tests/testautonatv2.nim

```diff
@@ -0,0 +1,120 @@
+{.used.}
+
+# Nim-Libp2p
+# Copyright (c) 2025 Status Research & Development GmbH
+# Licensed under either of
+#  * Apache License, version 2.0, ([LICENSE-APACHE](LICENSE-APACHE))
+#  * MIT license ([LICENSE-MIT](LICENSE-MIT))
+# at your option.
+# This file may not be copied, modified, or distributed except according to
+# those terms.
+
+import std/options
+import chronos
+import
+  ../libp2p/[
+    transports/tcptransport,
+    upgrademngrs/upgrade,
+    builders,
+    protocols/connectivity/autonatv2/core,
+      # nameresolving/nameresolver,
+      # nameresolving/mockresolver,
+  ],
+  ./helpers
+
+suite "AutonatV2":
+  teardown:
+    checkTrackers()
+
+  asyncTest "encode/decode messages":
+    # DialRequest
+    let dialReq = DialRequest(
+      addrs:
+        @[
+          MultiAddress.init("/ip4/127.0.0.1/tcp/4040").get(),
+          MultiAddress.init("/ip4/127.0.0.1/tcp/4041").get(),
+        ],
+      nonce: 42,
+    )
+    let encodedReq = dialReq.encode()
+    let decodedReq = DialRequest.decode(encodedReq).get()
+    check:
+      decodedReq.addrs == dialReq.addrs
+      decodedReq.nonce == dialReq.nonce
```

Comment:
```suggestion
    check decodedReq == dialReq
```

this is not really nitpick but the way you should check whole variables because it makes code more robust because changing variables in types you don't have to update tests. if property is removed then test update will be necessary, but if property is added will you really remember to check if that property is decoded/encoded?

also makes code much more simpler (smaller)

Reply:
100%